File: TUF/Numbers/CheckAbundantNumber.ts
function checkIsAbundantNumber(num: number): boolean {
    let sum: number = 0;
    let dupNum: number = num;
    while (num !== 0) {
        let rem: number = num % 10;
        sum += rem;
        num = Math.floor(num / 10);
    }
    return sum < dupNum;
}

let num: number = 18;
console.log(`${num} is an abundant number: ${checkIsAbundantNumber(num) ? "Yes" : "No"}`);

export { };

File: TUF/Numbers/CheckAbundantNumber.java
import java.util.Scanner;

public class CheckAbundantNumber {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the number to check: ");
        int num = input.nextInt();
        input.close();
        System.out.println(checkIsAbundantNumber(num));
    }

    private static boolean checkIsAbundantNumber(int num) {
        int sum = 0;
        int dupNum = num;
        while (num != 0) {
            int rem = num % 10;
            sum += rem;
            num = num / 10;
        }
        return sum < dupNum;
    }
}
